irods/test/access_test.py

```diff
@@ -332,6 +335,37 @@ def test_ses_acls_data_and_collection_395_396(self):
             self.alice.remove()
             self.team.remove()
 
+    def test_removed_user_does_not_affect_raw_ACL_queries__431(self):
+        user_name = "testuser"
+        session = self.sess
+        try:
+            # create user and collection
+            user = session.users.create(user_name, 'rodsuser')
+            coll_path = "/{0.zone}/home/test".format(session)
+            coll = session.collections.create(coll_path)
+
+            # give user access to collection
+            acces = iRODSAccess('read', coll.path, user.name)
+            session.acls.set(acces)
+
+            # we can get permissions from collection
+            perms = session.acls.get(coll)
+
+            # now we remove the user and try again
+            user.remove()
+
+            # The following line caused a KeyError prior to the issue #431 fix,
+            # as already-deleted users' IDs were being returned in the raw ACL queries.
+            # It appears iRODS as of 4.2.11 and 4.3.0 does not purge R_OBJT_ACCESS of old
```

Comment:
Have we confirmed that this works in previous releases?

Reply:
Not yet, will make to run tests with some older servers.

Reply:
> Have we confirmed that this works in previous releases?

Is there a particular reason that it might not work for some older ones?

Reply:
This comment suggests that the issue only appeared in 4.2.11 and 4.3.0 and I think I was just wondering if we knew this to be the case.

Reply:
I misconveyed then. I've tested as far back as  server version 4.2.7 and still found the issue irods/irods#6921 to be present.  It only came up this late because the (legacy) `session.permissions.get` interface doesn't poll for raw permissions by default. (ie attempting to match the listing of `ils -A` exactly), and the new interface `session.acls.get` does.

Reply:
Still will test, however.